src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -2829,6 +2966,38 @@ static void Main()
         CompileAndVerify(source, expectedOutput: "c5").VerifyDiagnostics();
     }
 
+    [Fact]
+    public void RefReadonlyParameter_OverloadResolution_04()
+    {
+        var source = """
+            interface I1 { }
+            interface I2 { }
+            class C
+            {
+                static string M1(I1 o, in int i) => "1" + i;
+                static string M1(I2 o, ref readonly int i) => "2" + i;
```

Comment:
>static string M1(I2 o, ref readonly int i) => "2" + i; [](http://example.com/codeflow?start=16&length=54)

Briefly discussed offline. It feels like overloading like this shouldn't be allowed, but the speclet doesn't say that.

Reply:
I think we said we will add a betterness rule - note that the first parameter type is different, and this is allowed for `ref`/`in` today, for example.